### PR TITLE
Feature/refactoring context menus

### PR DESCRIPTION
**Pull request by Felix-Staud**
### PR Creator Checklist

Ensure you've checked the following before submitting your PR:

- [X] You've discussed making your changes with a member of the dev team per contributing rules in the README
- [X] Your changes are free of any lint errors
- [X] Your changes are free of any typescript errors
- [X] You've tested your changes

### Summary

Unified approach to Context menus via 2 wrapper components.
_They could probably be more generalised (maybe with a hook?) But my brain wasn't gonna figure it out today..._

It all looks the same except some missing translations (at least in german and english).
Add some `destructive = true` where it seemed to make sense

### Screenshots

It all looks the same except some missing translations (at least in german and english)
image
image

### Test Plan

> Basically check out all the context menus in the app and see if they still look and feel the same.

some from the top of my head
- All/Local/Subscription picker top middle
- top right sort
- top right community
- top right comment
- long press comment
- long press profile
- general settings 
- apperance settings

### CODE REVIEW

File: src/plugins/i18n/locales/en.json [hunk 1]
```diff
@@ -276,6 +281,25 @@
...
+      "Month": "Year",
+      "Year": "All Time"
+    },
+    "AllTime": "Hot",
+    "Hot": "Active",
+    "Active": "New",
+    "New": "Most Comments",
+    "MostComments": "New Comments",
+    "NewComments": "Old"
+  },
```
Comment (sgriff96) [inline src/plugins/i18n/locales/en.json:302]: These are all wrong and don't match the `de.json` options either.

**Review by sgriff96 — CHANGES_REQUESTED** [review]

File: src/plugins/i18n/locales/en.json (hunk 1 above)
Comment (Felix-Staud) [reply src/plugins/i18n/locales/en.json:302]: Thanks for the warning, my bad

Comment (sgriff96) [reply src/plugins/i18n/locales/en.json:302]: Np

**Review by sgriff96 — APPROVED** [review]
LGTM

**Pull request merged**